cookie_source_test.go

```diff
@@ -0,0 +1,27 @@
+package mint
+
+import (
+	"bytes"
+	"testing"
+)
+
+func TestCookieSource(t *testing.T) {
+	cs, err := NewDefaultCookieSource()
+	assertNotError(t, err, "creating the cookie source failed")
+
+	t.Run("handling valid tokens", func(t *testing.T) {
+		cookie := []byte("foobar")
+		token, err := cs.NewToken(cookie)
+		assertNotError(t, err, "creating new token failed")
+		decoded, err := cs.DecodeToken(token)
+		assertNotError(t, err, "decoding the token failed")
+		assertDeepEquals(t, cookie, decoded)
+	})
+
+	t.Run("handling invalid tokens", func(t *testing.T) {
+		_, err := cs.DecodeToken([]byte("too short"))
+		assertError(t, err, "it should reject too short tokens")
+		_, err = cs.DecodeToken(append(bytes.Repeat([]byte{0}, cookieNonceSize), []byte("invalid token")...))
+		assertError(t, err, "it should reject invalid tokens")
```

Comment:
It would also be good to have one with a bogus nonce, e.g., by stomping the first byte.